### PR TITLE
[14.0][ADD] account_payment_order_lock_draft

### PR DESCRIPTION
**Pull request by kaynnan**
This module adds functionality to validate, during the draft function, if a user attempts to reset an invoice to draft status using the "Reset to Draft" button. The system checks if the invoice is associated with a Payment Order. This feature enhances security by preventing unintended modifications to invoices that are linked to Payment Order.

cc @marcelsavegnago @douglascstd @WesleyOliveira98 @Matthwhy

### CODE REVIEW

**Comment by kaynnan** [comment]
ping @OCA/accounting-maintainers

**Review by WesleyOliveira98 — APPROVED** [review]
LGTM

**Comment by kaynnan** [comment]
ping @OCA/accounting-maintainers

**Comment by kaynnan** [comment]
The latest commit added sudo to the invoice search process, ensuring users not in the Payment Order group don't receive incorrect warnings about group requirements. This prevents conflicts and ensures proper functionality.

**Review by marcelsavegnago — APPROVED** [review]

**Comment by kaynnan** [comment]
ping @OCA/accounting-maintainers

**Comment by kaynnan** [comment]
ping @OCA/banking-maintainers

File: account_payment_order_restriction_account_move/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+To use this module:
```
Comment (etobella) [inline account_payment_order_restriction_account_move/readme/USAGE.rst]: For this USAGE, you don't need to write it :wink:

**Review by etobella — COMMENTED** [review]
I like the idea of the module, but the name is something weird... I know it is always hard, what do you think something like:

- `account_payment_order_restrict_draft`
- `account_payment_order_lock_draft`

**Comment by kaynnan** [comment]
> I like the idea of the module, but the name is something weird... I know it is always hard, what do you think something like:
> 
>     * `account_payment_order_restrict_draft`
> 
>     * `account_payment_order_lock_draft`

I believe the name account_payment_order_lock_draft is good, I will make the changes and upload the commit

File: account_payment_order_restriction_account_move/readme/USAGE.rst (hunk 1 above)
Comment (etobella) [reply account_payment_order_restriction_account_move/readme/USAGE.rst]: Remove the file, you just kept an empty file :smile:

**Review by etobella — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**